### DISCUSSION TITLE
Add pip install package as quick option when package not found

### DISCUSSION THREAD

**Discussion by AAraKKe** [microsoft/pylance-release#2687] [Enhancement]
I have been searching for this for some time but I cannot find anything so either I am pretty bad at searching or this might have never been proposed, both of them very weird.

I cannot seem to find a way of making this work so I am guessing there is no option for this but it would be great if we could have an option of auto-installing packages as a quick action when not found. For example, imagine that I write something like:

```python
engine = sqlalchemy.create_engine(...)
```

As expected I get an error saying that sqlalchemy is not found. It would be great if I could just select a quick action to install that package into the current environment. Or from an import statement.

Is this something you are willing to work on, or something already exists and I am not able to find?

Thanks!

**Comment by karthiknadig** [comment]
Thank you for the suggestion! We have marked this issue as "needs decision" to make sure we have a conversation about your idea. We plan to leave this feature request open for at least a month to see how many 👍 votes the opening comment gets to help us make our decision.

**Comment by AAraKKe** [comment]
Hi! Sorry, these past weeks have been a nightmare at work 😅 I just saw your answer. Thanks a lot for considering this!

**Comment by savannahostrowski** [comment]
Thanks for the feature request. We'll have a look at this and follow up here when we have an update.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
